src/main/daml/Daml/Finance/Interface/Lifecycle/Rule/Lifecycle.daml

```diff
@@ -25,7 +26,7 @@ data View = View
 interface Lifecycle where
   viewtype V
 
-  evolve : Evolve -> ContractId Lifecycle -> Update (ContractId Lifecycle, [ContractId Effect.I])
+  evolve : Evolve -> Update (ContractId Instrument.I, [ContractId Effect.I])
```

Comment:
I think that the right output type here would be `Optional (ContractId Effect.I)` rather than `[ContractId Effect.I].

`None` if no effect is produced (because there is nothing happening in lifecycling or `Some effect`.